framework/13-inch/12th-gen-intel/default.nix

```diff
@@ -1,52 +1,61 @@
-    # Workaround iGPU hangs
-    # https://discourse.nixos.org/t/intel-12th-gen-igpu-freezes/21768/4
-    "i915.enable_psr=1"
-  ];
-
-  boot.blacklistedKernelModules = [ 
-    # This enables the brightness and airplane mode keys to work
-    # https://community.frame.work/t/12th-gen-not-sending-xf86monbrightnessup-down/20605/11
-    "hid-sensor-hub"
-    # This fixes controller crashes during sleep
-    # https://community.frame.work/t/tracking-fn-key-stops-working-on-popos-after-a-while/21208/32
-    "cros_ec_lpcs"
-  ];
-  
-  # Further tweak to ensure the brightness and airplane mode keys work
-  # https://community.frame.work/t/responded-12th-gen-not-sending-xf86monbrightnessup-down/20605/67
-  systemd.services.bind-keys-driver = {
-    description = "Bind brightness and airplane mode keys to their driver";
-    wantedBy = [ "default.target" ];
-    after = [ "network.target" ];
-    serviceConfig = {
-      Type = "oneshot";
-      User = "root";
-    };
-    script = ''
-      ls -lad /sys/bus/i2c/devices/i2c-*:* /sys/bus/i2c/drivers/i2c_hid_acpi/i2c-*:*
-      if [ -e /sys/bus/i2c/devices/i2c-FRMW0001:00 -a ! -e /sys/bus/i2c/drivers/i2c_hid_acpi/i2c-FRMW0001:00 ]; then
-        echo fixing
-        echo i2c-FRMW0001:00 > /sys/bus/i2c/drivers/i2c_hid_acpi/bind
-        ls -lad /sys/bus/i2c/devices/i2c-*:* /sys/bus/i2c/drivers/i2c_hid_acpi/i2c-*:*
-        echo done
-      else
-        echo no fix needed
-      fi
-    '';
-  };
+  config = lib.mkMerge [
+    {
+      hardware.intelgpu.loadInInitrd = lib.versionOlder config.boot.kernelPackages.kernel.version "6.2";
+    }
+    # https://community.frame.work/t/tracking-hard-freezing-on-fedora-36-with-the-new-12th-gen-system/20675/391
+    (lib.mkIf (lib.versionOlder config.boot.kernelPackages.kernel.version "6.2") {
+      boot.kernelParams = [ 
+        # Workaround iGPU hangs
+        # https://discourse.nixos.org/t/intel-12th-gen-igpu-freezes/21768/4
+        "i915.enable_psr=1" 
+      ];
+    })
+    (lib.mkIf (lib.versionOlder config.boot.kernelPackages.kernel.version "6.8") {
+      boot.blacklistedKernelModules = [ 
+        # This enables the brightness and airplane mode keys to work
+        # https://community.frame.work/t/12th-gen-not-sending-xf86monbrightnessup-down/20605/11
+        "hid-sensor-hub"
+        # This fixes controller crashes during sleep
+        # https://community.frame.work/t/tracking-fn-key-stops-working-on-popos-after-a-while/21208/32
+        (lib.mkIf (config.hardware.framework.enableKmod == false) "cros_ec_lpcs")
+      ];
 
-  # Alder Lake CPUs benefit from kernel 5.18 for ThreadDirector
-  # https://www.tomshardware.com/news/intel-thread-director-coming-to-linux-5-18
-  boot.kernelPackages = lib.mkIf (lib.versionOlder pkgs.linux.version "5.18") (lib.mkDefault pkgs.linuxPackages_latest);
+      # Combined with acpi_osi from framework-intel it increases the idle power-usage from 0.X Watt to up to 2.X Watt in my test (SebTM - 04/2024)
```

Comment:
Perhaps this is me misreading the comment, but is this saying that *removing* the following `boot.kernelParams` option as well as the acpi_osi will increase the idle power usage?

Reply:
It means, if both options this and the acpi_osi from framework-intel on new kernel-versions (e.g. 6.8) are enabled it increases instead of optimizing the power-usage.

See my redone test-results I will post as comment in a few minutes ✌🏻